### PR TITLE
Feat/email password auth

### PR DESCRIPTION
**Pull request by cathytham**
# This is the PR template! please make sure to self review your PR 
# Short description of what your pr does
Log In/ Sign Up forms that will connect to the backend and will register or log in a user. 
Displays an error message if there is one.

**Here are some screenshot if the Log In and Sign up**
image0
image1

## TODO
On ligne 236-261 of LogIn.js, code commented out to be use in the future.
On ligne 250-275 of SignUp.js, code commented out to be use in the future.
On ligne 68-76 of App.js, code commented out to be use in the future.

## Self Review process
- [x] no console logs
- [x] comments EXTREMELY IMPORTANT
- [ ] what needs to be in a configuration file is in a configuration file
- [x] does not touch unesseary components
- [ ] PR is for one purpose
- [x] code is formatted

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+  "compilerOptions": {},
```
Comment (stefanoScalzo) [inline tsconfig.json]: what is this file needed for ?

File: src/edit-profile/terms.ts [hunk 2]
```diff
@@ -0,0 +1,430 @@
...
+Interpretation and Definitions
+Interpretation
+The words of which the initial letter is capitalized have meanings defined under the following conditions. The following definitions shall have the same meaning regardless of whether they appear in singular or in plural.
+
+Definitions
+For the purposes of this Privacy Policy:
+
+Account means a unique account created for You to access our Service or parts of our Service.
+
+Affiliate means an entity that controls, is controlled by or is under common control with a party, where "control" means ownership of 50% or more of the shares, equity interest or other securities entitled to vote for election of directors or other managing authority.
```
Comment (stefanoScalzo) [inline src/edit-profile/terms.ts]: i thnk we should only be using js files in react

File: src/components/styles.js [hunk 3]
```diff
@@ -0,0 +1,201 @@
+import { StyleSheet, Dimensions } from "react-native";
+/**
+ * This is the style sheet for the log in/sign up forms.
+ */
+const styles = StyleSheet.create({
+  textForgotPass: {
+    fontSize: 19,
+    fontWeight: "bold",
```
Comment (stefanoScalzo) [inline src/components/styles.js]: lets put this in a global folder and call it global-styles

File: src/components/styles.js [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+    color: "darkviolet",
+  },
+  buttonForgotPass: {
+    alignItems: "center",
+    justifyContent: "center",
+    borderColor: "darkviolet",
+    borderWidth: 1,
+    backgroundColor: "white",
+    borderRadius: 100,
+    marginTop: 10,
```
Comment (stefanoScalzo) [inline src/components/styles.js]: any margin, paddings, almost any number that is used in more than one spot should be a constant

Comment (stefanoScalzo) [inline src/components/styles.js]: also, any margin or padding should not be in a general style but in a component-specific style

File: tsconfig.json (hunk 1 above)
Comment (cathytham) [reply tsconfig.json]: It is use for the typescript files.

File: src/components/SignUp/SignUp.js [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+      },
+    ]);
+  };
+
+  /**
+   * async helper function to register the user
+   * @param {*} data is the values entered by the user
+   */
+  async function agreeOption(data) {
+    let datePieces = data.dob.split("-");
```
Comment (stefanoScalzo) [inline src/components/SignUp/SignUp.js]: lets rename this to something more clear, same for data, data from what

File: src/components/SignUp/SignUp.js [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(newUser),
+      });
+      let responseJSON = await response.json();
+      if (responseJSON["status"] == "valid") {
+        await SecureStore.setItemAsync("jwt", responseJSON["data"]["jwt"]);
+        updateTokenInDatabase();
+        console.log("logged in");
```
Comment (stefanoScalzo) [inline src/components/SignUp/SignUp.js]: no console logs

File: src/components/SignUp/SignUp.js [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+      } else {
+        setError("There was an error " + responseJSON["error"]);
```
Comment (stefanoScalzo) [inline src/components/SignUp/SignUp.js]: does this error display the message in the component. ?

File: src/components/SignUp/SignUp.js [hunk 8]
```diff
@@ -0,0 +1,246 @@
...
+        isValid,
+      }) => (
+        <View>
+          <DatePicker
+            style={styles.datePicker}
+            date={values.dob}
+            mode="date"
+            placeholder="Date of Birth"
+            format="YYYY-MM-DD"
+            // minDate="1920-05-01"
```
Comment (stefanoScalzo) [inline src/components/SignUp/SignUp.js]: remove comments

File: src/components/SignUp/SignUp.js [hunk 9]
```diff
@@ -0,0 +1,246 @@
...
+                display: "none",
+              },
+              dateInput: {
+                borderWidth: 0,
+                alignItems: "flex-start",
+                justifyContent: "flex-end",
+              },
+              dateText: {
+                color: "white",
+                fontSize: 16,
```
Comment (stefanoScalzo) [inline src/components/SignUp/SignUp.js]: constant

File: src/components/SignUp/SignUp.js (hunk 6 above)
Comment (cathytham) [reply src/components/SignUp/SignUp.js]: You told me to add the console.log for this part, to make sure the user is actually log in. Do I still remove it ?

File: src/components/SignUp/SignUp.js (hunk 7 above)
Comment (cathytham) [reply src/components/SignUp/SignUp.js]: yes, it does display "There was an error [object object]". However, the responseJSON returns object. Therefore, I wanted to remove "responseJSON["error"]".

File: tsconfig.json (hunk 1 above)
Comment (stefanoScalzo) [reply tsconfig.json]: there shouldnt be typescript files though, if there is they should be changed to js files

File: src/components/SignUp/SignUp.js (hunk 6 above)
Comment (stefanoScalzo) [reply src/components/SignUp/SignUp.js]: yes just for the start, but now that we have other information we can remove it

File: src/components/SignUp/SignUp.js (hunk 7 above)
Comment (stefanoScalzo) [reply src/components/SignUp/SignUp.js]: it needs to display the appropriate error though, find out what [object][object] is and then display the proper error

File: tsconfig.json (hunk 1 above)
Comment (cathytham) [reply tsconfig.json]: alright

File: src/components/SignUp/SignUp.js (hunk 7 above)
Comment (cathytham) [reply src/components/SignUp/SignUp.js]: Okay, I did try to get the right error but couldn't find a solution. I'll try again and will ask for help if necessary. Thank you

File: src/components/SignUp/SignUp.js (hunk 6 above)
Comment (stefanoScalzo) [inline src/components/SignUp/SignUp.js]: remove

Comment (cathytham) [reply src/components/SignUp/SignUp.js]: This is strange, I don't have the console.log on my side

File: src/components/LogIn/LogIn.js [hunk 10]
```diff
@@ -0,0 +1,258 @@
...
+      });
+      let responseJSON = await response.json();
+      if (responseJSON["status"] == "valid") {
+        await SecureStore.setItemAsync("jwt", responseJSON["data"]["jwt"]);
+        updateTokenInDatabase();
+      } else {
+        if (responseJSON["data"]) {
+          setError(responseJSON["data"]["message"]);
+        } else {
+          setError(responseJSON["error"]["code"]);
```
Comment (stefanoScalzo) [inline src/components/LogIn/LogIn.js:50]: you should check the code and send a more redable message

File: src/global/global-styles.js [hunk 11]
```diff
@@ -0,0 +1,186 @@
...
+const imgHeight=100;
+const imgWidth=100;
+const btnFontSize=19;
+const inputFontSize=16;
+const letterSpacing=0.25;
+const borderRadius=100;
+const btnmarginBottom=10;
+const borderWidth=1;
+const navBtnPaddingLeftRight="6%";
+const navBtnPaddingTopBottom="3%";
```
Comment (stefanoScalzo) [inline src/global/global-styles.js]: all these numbers should be in the global constant 

btnmarginBottom should be marginSmall

File: src/global/global-styles.js [hunk 12]
```diff
@@ -0,0 +1,186 @@
+import { StyleSheet, Dimensions } from "react-native";
+/**
+ * This is the style sheet for the log in/sign up forms.
+ */
+
+const containerWidth="100%";
+const imgHeight=100;
+const imgWidth=100;
+const btnFontSize=19;
```
Comment (stefanoScalzo) [inline src/global/global-styles.js]: lets use full words

File: src/global/global-styles.js [hunk 13]
```diff
@@ -0,0 +1,186 @@
...
+    fontSize: btnFontSize,
+    fontWeight: "bold",
+    letterSpacing: letterSpacing,
+    color: "darkviolet",
+  },
+
+  buttonSignUp: {
+    alignItems: "center",
+    justifyContent: "center",
+    borderColor: "darkviolet",
```
Comment (stefanoScalzo) [inline src/global/global-styles.js]: lets make this a constant

File: src/components/LogIn/LogIn.js (hunk 10 above)
Comment (cathytham) [reply src/components/LogIn/LogIn.js:50]: I didn't find a specific error code for this. The 2 possible errors that I found were already caught in the responseJSON["data"]["message"].

File: App.js [hunk 14]
```diff
@@ -51,17 +55,27 @@ export default function App() {
   };
 
   return (
-    <View style={styles.container}>
-      {user ? (
+    <LinearGradient
+      colors={["darkviolet", "#1D001D"]}
```
Comment (stefanoScalzo) [inline App.js]: constant

File: src/global/global-styles.js [hunk 15]
```diff
@@ -0,0 +1,135 @@
...
+  textSignIn: {
+    color: "white",
+    fontWeight: "bold",
+    fontSize: buttonFontSize,
+  },
+
+  buttonSignIn: {
+    alignItems: "center",
+    justifyContent: "center",
+    backgroundColor: "darkviolet",
```
Comment (stefanoScalzo) [inline src/global/global-styles.js]: constant

File: src/global/global-styles.js [hunk 16]
```diff
@@ -0,0 +1,135 @@
...
+  },
+  image: {
+    width: imageWidth,
+    height: imageHeight,
+    borderRadius: 50,
+  },
+  //styles for KeyboardAvoidingInputs.js
+  keyboardAvoidingViewStyle: {
+    flex: 1,
+    width: Dimensions.get("window").width,
```
Comment (stefanoScalzo) [inline src/global/global-styles.js]: use screen, and try it out its usually more reliable than window

File: src/components/SignUp/SignUp.js [hunk 17]
```diff
@@ -0,0 +1,281 @@
...
+              {
+                borderColor: globalConstant.blackColor,
+                borderWidth: globalConstant.formButtonBorderWidth,
+                backgroundColor: globalConstant.whiteColor,
+              },
+            ]}
+          >
+            <Text style={styles.textAG}>Sign In with Apple</Text>
+          </TouchableOpacity>
+          <TouchableOpacity
```
Comment (stefanoScalzo) [inline src/components/SignUp/SignUp.js:263]: you can remove these since were only going to have the sign in with google
